src/compiler/emitter.ts

```diff
@@ -1592,13 +1592,13 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 /// these emit into an object literal property name, we don't need to be worried
                 /// about keywords, just non-identifier characters
                 function emitAttributeName(name: Identifier) {
-                    if (/[A-Za-z_]+[\w*]/.test(name.text)) {
-                        write("\"");
+                    if (/^[A-Za-z_]+[\w]*$/.test(name.text)) {

```

Comment:
Re 3, I don't know. The failure mode here is that we quote something that didn't *need* to be quoted, which isn't really a problem per se